gpflow/test_utils.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+
+from .base import AnyNDArray
+from .kernels import Kernel
+
+
+def assert_psd_matrix(A: AnyNDArray, tol: float = 1e-12) -> None:
+    """
+    Assert whether `A` is a positive semi-definite matrix, based on the
+    smallest eigenvalue being non-negative to tolerance `tol`.
+
+    :param A: the matrix to check for positive-semi-definiteness
+    :param tol: numeric tolerance to accept if the smallest eigenvalue is very
+        slightly negative.
+    """
+    assert np.linalg.eigvals(A).min() > -tol, "test for positive semi definite matrix"
+
+
+def test_kernel(kernel: Kernel, X: AnyNDArray, X2: AnyNDArray) -> None:
```

Comment:
I'm not sure it is good that this is called `test_kernel`. I think `pytest` thinks that anything that starts with `test_` is a unit test, which this obviously isn't. If we can find a name that doesn't start with `test_` we might avoid some confusion?

Reply:
```suggestion
def check_kernel_interface(kernel: Kernel, X: AnyNDArray, X2: AnyNDArray) -> None:
```